src/components/Inputs/GenericInput.tsx

```diff
@@ -0,0 +1,143 @@
+import styled from '@emotion/styled';
+import React from 'react';
+import { pxToRem } from '../../utils';
+
+export type GenericLabelPositions = 'top' | 'right' | 'bottom' | 'left';
+export type GenericTypes = 'text' | 'email' | 'password';
+
+export interface GenericInputProps {
+  readonly inputType: GenericTypes;
+  readonly label?: string;
+  readonly labelPosition?: GenericLabelPositions;
+  readonly gapSize?: number;
+  readonly fontSize?: number;
+  readonly fontColor?: string;
+  readonly placeholder?: string;
+  readonly placeholderColor?: string;
+  readonly boxShadowColor?: string;
+  readonly borderColor?: string;
+  readonly focusBorderColor?: string;
+  readonly focusBorderWidth?: number;
+  readonly width?: number;
+  readonly height?: number;
+  readonly minTextLength?: number;
+  readonly maxTextLength?: number;
+  readonly spellcheck?: boolean;
+  readonly pattern?: string;
+  readonly disabled?: boolean;
+  readonly required?: boolean;
+  readonly onChange?: React.ChangeEventHandler<HTMLInputElement>;
+}
+
+export const GenericInput: React.FC<GenericInputProps> = ({
+  inputType = 'tel',
+  label = 'Generic label',
+  labelPosition = 'left',
+  gapSize,
+  fontSize,
+  fontColor,
+  placeholder = 'Enter text',
+  placeholderColor,
+  boxShadowColor,
+  borderColor,
+  focusBorderColor,
+  focusBorderWidth,
+  width,
+  height,
+  minTextLength,
+  maxTextLength,
+  spellcheck,
+  pattern,
+  onChange,
+  required,
+  disabled,
+}) => (
+  <LabelPasswordInputContainer labelPosition={labelPosition} gapSize={gapSize}>
+    <label>{label}</label>
```

Comment:
Shouldn't we have a `for` attribute on this `label`?

Reply:
According to MDN, 
"To explicitly associate a <label> element with an <input> element, you first need to add the id attribute to the <input> element. Next, you add the for attribute to the <label> element, where the value of for is the same as the id in the <input> element." 
Currently none of our inputs have default, unique ids.  Maybe it would make sense to add these two attributes to all of the components in the ui kit?